src/Network/Network/NetworkSecurityPerimeter.Autorest/README.md

```diff
@@ -0,0 +1,187 @@
+
+## Info
+- Modifiable: yes
+- Generated: all
+- Committed: yes
+- Packaged: yes
+
+---
+## Detail
+This module was primarily generated via [AutoRest](https://github.com/Azure/autorest) using the [PowerShell](https://github.com/Azure/autorest.powershell) extension.
+
+## Module Requirements
+- [Az.Accounts module](https://www.powershellgallery.com/packages/Az.Accounts/), version 2.7.5 or greater
+
+## Authentication
+AutoRest does not generate authentication code for the module. Authentication is handled via Az.Accounts by altering the HTTP payload before it is sent.
+
+## Development
+For information on how to develop for `Az.NetworkSecurityPerimeter`, see [how-to.md](how-to.md).
+<!-- endregion -->
+
+### AutoRest Configuration
+> see https://aka.ms/autorest
+
+``` yaml
+require:
+# readme.azure.noprofile.md is the common configuration file
+# - ../../../../tools/SwaggerCI/readme.azure.noprofile.md
+  - $(this-folder)/../../../readme.azure.noprofile.md
+input-file:
+# You need to specify your swagger files here.
+  - https://github.com/kaushal087/azure-rest-api-specs/blob/nsp_powershell/specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkSecurityPerimeter.json
+#  - C:\repo\azure-rest-api-specs/specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkSecurityPerimeter.json
+# If the swagger has not been put in the repo, you may uncomment the following line and refer to it locally
+# - (this-folder)/relative-path-to-your-swagger 
+
+# For new RP, the version is 0.1.0
+module-version: 0.1.0
+# Normally, title is the service name
+title: NetworkSecurityPerimeter
+subject-prefix: $(service-name)
+sanitize-names: true
+
+# If there are post APIs for some kinds of actions in the RP, you may need to 
+# uncomment following line to support viaIdentity for these post APIs
+# identity-correction-for-post: true
+
+directive:
+  # Following is two common directive which are normally required in all the RPs
+  # 1. Remove the unexpanded parameter set
+  # 2. For New-* cmdlets, ViaIdentity is not required, so CreateViaIdentityExpanded is removed as well
+#  - where:
+#      variant: ^Create$|^CreateViaIdentityExpanded$
+#    remove: true
+# Remove the set-* cmdlet
+  - where:
+      verb: Set
+    remove: true
+  - where:
+      variant: ^Update$|^UpdateViaIdentity$|^UpdateViaIdentityExpanded$
+    remove: true
+  # Hide CreateViaIdentity for customization
+  - where:
+      variant: ^CreateViaIdentity$
+    hide: true
+  - where:
+      subject: NetworkSecurityPerimeter
+    set:
+      subject-prefix: ''
+  - where:
+      subject: NspProfile
+    set:
+      subject: Profile
+  - where:
+      subject: NspAccessRule
+    set:
+      subject: AccessRule
+  - where:
+      subject: NspAssociation
+    set:
+      subject: Association
+
+# Parameter Update
+# NSP
+  - where:
+      subject: NetworkSecurityPerimeter
+      parameter-name: Name
+    set:
+      parameter-name: Name
```

Comment:
parameter-name in the `set` statement can be omitted when it's not changed
```suggestion
```

Reply:
Removed.